.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,34 @@
+name: Doxygen GitHub Pages Deploy Action
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-documentation:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+          submodules: true
+
+    # Build the HTML documentation
+    - name: Doxygen Action
+      uses: mattnotmitt/doxygen-action@v1.9.4
+      with:
+          doxyfile-path: docs/Doxyfile
+          working-directory: .
+          additional-packages: perl build-base biblatex ghostscript
+
+    # Deploy the HTML documentation to GitHub Pages
+    - name: GH Pages Deployment
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.ACTIONS_SECRET_KURPICZ }}
```

Comment:
I don't really know how these secrets work, but this name doesn't seem right. Can't it be called `ACTIONS_SECRET_DOXYGEN` or something like that?

Reply:
`GITHUB_TOKEN` doesn't work?

Reply:
`GITHUB_TOKEN` doesn't work?

Reply:
Is there a default GitHub-Token? If so, I am not aware of it. I'll look into that.

Reply:
It's the one I found when looking for these kind of things in workflows:
https://docs.github.com/en/actions/security-guides/automatic-token-authentication